pkg/pushsync/pushsync.go

```diff
@@ -270,23 +272,42 @@ type pushResult struct {
 	attempted bool
 }
 
-func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk, retryAllowed bool, origin swarm.Address) (*pb.Receipt, error) {
+func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk, origin swarm.Address) (*pb.Receipt, error) {
 	span, logger, ctx := ps.tracer.StartSpanFromContext(ctx, "push-closest", ps.logger, opentracing.Tag{Key: "address", Value: ch.Address().String()})
 	defer span.Finish()
 	defer ps.skipList.PruneExpired()
 
+	originated := false
+	if origin.IsZero() {
+		originated = true
+	}
+
+	// Set up channel and timer for preemptive storage establishment as in-neighborhood forwarder
+	var sendReceipt <-chan time.Time
+	chunkInNeighbourhood := ps.topologyDriver.IsWithinDepth(ch.Address())
+	if chunkInNeighbourhood {
+		sendReceipt = time.After(sendReceiptDelay)
+	}
+
 	var (
 		skipPeers      []swarm.Address
 		allowedRetries = 1
-		resultC        = make(chan *pushResult)
+		resultC        = make(chan *pushResult, 1)
```

Comment:
Why buffered channel here?

Reply:
it was intended for the concurrent storage establishment / pending push towards peer case, so that they can both write to this channel without blocking the other